cache/lib/cache/s3_transfer.ex

```diff
@@ -0,0 +1,21 @@
+defmodule Cache.S3Transfer do
+  @moduledoc """
+  Schema for pending S3 transfers.
+
+  This table tracks artifacts that need to be uploaded to or downloaded from S3.
+  It replaces per-request Oban job insertion with a simpler queue model that
+  avoids SQLite contention under bursty load.
```

Comment:
is there an issue in oban? I'd expect them to be able this scenario more gracefully by having a queue. If there's not, might be worth to creating one, would love to offload this to them. We can also consider getting [Oban Pro](https://oban.pro/), but I don't think Oban Pro would solve this particular issue, but might be worth reaching out.